### PR TITLE
Docarray vectorstore

### PR DESCRIPTION
**Pull request by jupyterjazz**

### CODE REVIEW

File: langchain/vectorstores/vector_store_from_doc_index.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    except ImportError:
+        raise ImportError(
+            "Could not import docarray python package. "
+            "Please install it with `pip install \"langchain[docarray]\"`."
+        )
+
+
+class VecStoreFromDocIndex(VectorStore):
+    doc_index: BaseDocIndex
+    doc_cls: Type[BaseDoc]
```
Comment (samsja) [inline langchain/vectorstores/vector_store_from_doc_index.py:39]: how can we change the `doc_cls` dynamically ?

**Review by samsja — COMMENTED** [review]
looks great, I have one tiny comment on the cls things

File: langchain/vectorstores/vector_store_from_doc_index.py (hunk 1 above)
Comment (jupyterjazz) [reply langchain/vectorstores/vector_store_from_doc_index.py:39]: child classes (in memory and hnsw) call `_get_doc_cls` with specific arguments

**Review by samsja — APPROVED** [review]
lgtm

**Review by JoanFM — APPROVED** [review]

**Pull request merged**